src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -5094,8 +5059,47 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
         )
         silentResult match {
           case SilentResultValue(fun1) =>
+            def transformRassoc(applied: Tree): Tree = {
+              val Apply(fn, arg :: Nil) = applied: @unchecked
+              val isByName = fn.symbol.paramLists.head.head.isByNameParam
```

Comment:
maybe we could go for a more cautious approach here to avoid crashes, just in case.